src/event_log.cpp

```diff
@@ -168,8 +169,9 @@ void EventLog::enumerate(uint32_t index, std::function<void(String time, EventTy
         double temperature = json["tp"];
         double temperatureMax = json["tm"];
         uint8_t divertMode = json["dm"];
+        uint8_t shaper = json["sh"];
```

Comment:
Have you tested for backward compatibility with logs that do not have the shaper info?

Reply:
> I don't think we need additional APIs for enabling/disabling the shaper, this can entirely be done via the config. To my mind it is a safety feature so should be setup and enabled then left. I don't think it should have the same MQTT/HTTP APIs to 'easily' switch on/off from external sources that the divert mode does, but other than that I think this is looking really good, will do some testing , hopefully tomorrow.

I have some use case where it's needed. Sometime I override the shaper to let the car having max priority on my house load. Then the house heaters are shaping the current instead.

Therefore, I've made a quick live switch to override it without writing in the config. 



> Have you tested for backward compatibility with logs that do not have the shaper info?

not tested that.